lib/galaxy/jobs/runners/lsf.py

```diff
@@ -0,0 +1,240 @@
+"""
+Job control via the LSF.
+"""
+import logging
+import os
+import stat
+
+from galaxy import model
+from galaxy.jobs.runners import (
+    AsynchronousJobRunner,
+    AsynchronousJobState
+)
+from galaxy.jobs.runners.util.lsf import (
+    build_submit_description,
+    lsf_stop,
+    lsf_submit,
+    submission_params,
+    lsf_bjob
+)
+from galaxy.util import asbool
+
+log = logging.getLogger(__name__)
+
+__all__ = ('LSFJobRunner', )
+
+
+class LSFJobState(AsynchronousJobState):
+    def __init__(self, **kwargs):
+        """
+        Encapsulates state related to a job that is being run via the DRM and
+        that we need to monitor.
+        """
+        super(LSFJobState, self).__init__(**kwargs)
+        self.failed = False
+        self.user_log = None
+        self.user_log_size = 0
+
+
+class LSFJobRunner(AsynchronousJobRunner):
+    """
+    Job runner backed by a finite pool of worker threads. FIFO scheduling
+    """
+    runner_name = "LSFRunner"
+
+    def __init__(self, app, nworkers):
+        """Initialize this job runner and start the monitor thread"""
+        super(LSFJobRunner, self).__init__(app, nworkers)
+        self._init_monitor_thread()
+        self._init_worker_threads()
+
+    def queue_job(self, job_wrapper):
+        """Create job script and submit it to the DRM"""
+
+        # prepare the job
+        include_metadata = asbool(job_wrapper.job_destination.params.get("embed_metadata_in_job", True))
+        if not self.prepare_job(job_wrapper, include_metadata=include_metadata):
+            return
+
+        # get configured job destination
+        job_destination = job_wrapper.job_destination
+
+        # wrapper.get_id_tag() instead of job_id for compatibility with TaskWrappers.
+        galaxy_id_tag = job_wrapper.get_id_tag()
+
+        # get destination params
+        query_params = submission_params(prefix="", **job_destination.params)
+        container = None
+        universe = query_params.get('universe', None)
```

Comment:
This is very Condor specific and should be dropped I think.

Reply:
will drop the universe, I'm using query_params to build the execution file for LSF with its options.